Change 'size' props to 'variant' (#581)
primer/components: docs/content/TextInput.md

## Component props

> Props other than the ones listed here are forwarded to the underlying React `input` component.

Reviewer: ```suggestion
Native `<input>` attributes are forwarded to the underlying React `input` component and are not listed below.
```
